src/libraries/Native/Unix/System.Security.Cryptography.Native/pal_evp_cipher.c

```diff
@@ -332,3 +342,19 @@ const EVP_CIPHER* CryptoNative_EvpRC2Cbc()
 {
     return EVP_rc2_cbc();
 }
+
+const EVP_CIPHER* CryptoNative_EvpChaCha20Poly1305()
+{
+#if HAVE_OPENSSL_CHACHA20POLY1305
+    if (API_EXISTS(EVP_chacha20_poly1305))
+    {
+        return EVP_chacha20_poly1305();
+    }
+    else
+    {
+        return NULL;
+    }
+#else
+    return NULL;
+#endif
```

Comment:
nit, maybe the slimmer variant:
```suggestion
#endif
    return NULL;
```